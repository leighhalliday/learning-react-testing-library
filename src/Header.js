import React from "react";

export default function Header({ text }) {
  return (
    <header>
      <h1 data-testid="h1tag" className="fancy-h1">
        {text}
      </h1>
    </header>
  );
}
