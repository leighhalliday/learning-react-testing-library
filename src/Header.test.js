import React from "react";
import { render, cleanup } from "react-testing-library";
import "jest-dom/extend-expect";
import Header from "./Header";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<Header text="Hello!" />);
  expect(asFragment()).toMatchSnapshot();
});

it("inserts text in h1", () => {
  const { getByTestId, getByText } = render(<Header text="Hello!" />);

  expect(getByTestId("h1tag")).toHaveTextContent("Hello!");
  expect(getByText("Hello!")).toHaveClass("fancy-h1");
});
