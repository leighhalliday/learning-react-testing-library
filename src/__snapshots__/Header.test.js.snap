// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <header>
    <h1
      class="fancy-h1"
      data-testid="h1tag"
    >
      Hello!
    </h1>
  </header>
</DocumentFragment>
`;
